pkg/collect/host_subnetavailable.go

```diff
@@ -0,0 +1,287 @@
+package collect
+
+import (
+	"bytes"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"net"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	"github.com/apparentlymart/go-cidr/cidr"
+	"github.com/pkg/errors"
+	troubleshootv1beta2 "github.com/replicatedhq/troubleshoot/pkg/apis/troubleshoot/v1beta2"
+	"github.com/replicatedhq/troubleshoot/pkg/debug"
+)
+
+type SubnetStatus string
+
+const (
+	SubnetStatusAvailable     = "a-subnet-is-available"
+	SubnetStatusNoneAvailable = "no-subnet-available"
+)
+
+type SubnetAvailableResult struct {
+	CIDRRangeAlloc string `json:"CIDRRangeAlloc"`
+	DesiredCIDR    int    `json:"desiredCIDR"`
+	// If subnet-available, at least 1 of the DesiredCIDR size is available within CIDRRangeAlloc
+	Status SubnetStatus `json:"status"`
+}
+
+type CollectHostSubnetAvailable struct {
+	hostCollector *troubleshootv1beta2.SubnetAvailable
+	BundlePath    string
+}
+
+func (c *CollectHostSubnetAvailable) Title() string {
+	return hostCollectorTitleOrDefault(c.hostCollector.HostCollectorMeta, "Subnet Available")
+}
+
+func (c *CollectHostSubnetAvailable) IsExcluded() (bool, error) {
+	return isExcluded(c.hostCollector.Exclude)
+}
+
+func (c *CollectHostSubnetAvailable) Collect(progressChan chan<- interface{}) (map[string][]byte, error) {
+	procNetRoute, err := os.ReadFile("/proc/net/route")
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to read contents of /proc/net/route")
+	}
+
+	routes, err := parseProcNetRoute(string(procNetRoute))
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to parse /proc/net/route")
+	}
+	debug.Printf("Routes: %+v\n", routes)
+
+	// IPv4 only right now...
+	if c.hostCollector.DesiredCIDR < 1 || c.hostCollector.DesiredCIDR > 32 {
```

Comment:
`/0` is a valid mask unless we intentionally want to omit it

Reply:
Agreed, `/0` is a valid mask, in this instance I don't think we want to allow someone to select it though :) As that would basically be claiming the entirety of the valid IPv4 address space as an available subnet.

tbh I'd almost suggest we could lock this down to less than a `/8` too, but I'm open to leaving it fairly wide... feedback welcome on the "right" number there.

Reply:
Isn't the collector meant to be informational only? i.e it does not allow/deny anything, it just states facts, then the analyser enforces such constraints.

If we take this route, I suggest we document this somewhere, but I lean towards not having an opinion in the collector, rather in the analyser if need be.

Reply:
Technically the collector makes the decisions here... the analyzer just reports the results on this one. Which I agree does feel a bit counterintuitive.

The alternative approach would be the collector writes out all of the system routes, and then the analyzer does all of the logic we're doing in the collector today? The system routes are the main input component (in addition to the collector config attributes)...

Reply:
> The alternative approach would be the collector writes out all of the system routes

I think the current approach is how its meant to be.